chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -762,6 +778,131 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
             .collect();
         Ok(result)
     }
+
+    // Disconnect all blocks from the mainchain up until common ancestor of provided block
+    // and then connect all block in the new branch.
+    // All these operations are performed via `TransactionVerifier` without db modifications
+    // and the resulting delta is returned.
+    pub fn reorganize_in_memory(
```

Comment:
Please move this to a submodule. In the future we'll try to abstract this in a better way.

Reply:
done